### ISSUE TITLE
grafana url does not follow docs

### ISSUE THREAD

**Issue by stephen-dahl** [kiali/kiali#986]
**Describe the bug**
docs say that GRAFANA_URL defaults to grafana but it defaults to empty string in config/config.go#L238
also setting in the config file does not work as shown in docs
```
external_services:
  grafana:
    url: VALUE
```

**Versions used**
Kiali: 0.17
Istio: 1.1.0
Kubernetes flavour and version: eks

**Comment by stephen-dahl** [comment]
docs: https://github.com/kiali/kiali/tree/v0.17#environment-variables

**Comment by jmazzitelli** [comment]
Someone recently asked me about this -- here's the reply I sent:

Right now, Kiali cannot auto-detect Grafana or Jaeger so unless you set it yourself you won't have the URLs set. (This has been a sort spot for me for a while now :-) So you need to go into the Kiali configmap, edit it and fill in your Grafana and Jaeger URLs (if you have them installed and exposed to users), and then restart the Kiali pod.

But fear not! There is a PR [1] in the works that will at least attempt to auto-detect them when they are deployed and exposed... it won't detect them under 100% of all circumstances, but should auto-detect for many circumstances. This is not merged yet, so this feature isn't available to you.

Also, the new Kiali Operator will also auto-detect these at install time, so the configmap will have the settings set for you (if deployed on OpenShift with routes defined for them - that's what the Operator will look for). Again, not published so not available to you yet.

In short - edit the configmap manually and restart Kiali.

[1] https://github.com/kiali/kiali/pull/968

**Comment by stephen-dahl** [comment]
Mostly what I am bringing up with this ticket is that the docks say it defaults to a value but it really defaults to an empty string. also setting in the config map did not work for me. I had to set with env vars

**Comment by stephen-dahl** [comment]
correction, I got confused with GRAFANA_SERVICE having a default. but I was still unable to set the url in the config file.

**Comment by jmazzitelli** [comment]
config map should have this (after editing config map, restart the kiali pod):

```
external_services:
  grafana:
    url: http://your-grafana-url
```

**Comment by jmazzitelli** [comment]
FYI: These docs are going to change radically soon (once the operator PR goes in)... specifically, that environment variable section is going away. We are going to configure Kiali using a custom resource that gets handed to the kiali operator.

**Comment by stephen-dahl** [comment]
fair enough
